src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -99,34 +99,47 @@ bool SlaterDetBuilder::put(xmlNodePtr cur)
   xmlNodePtr curRoot=cur;
   xmlNodePtr BFnode;
   bool success=true, FastMSD=true;
-  std::string cname, tname;
+  std::string cname, tname,H5File;
   std::map<std::string,SPOSetBasePtr> spomap;
   bool multiDet=false;
+  bool ReadFromH5=false;
+  bool H5ReadOnce=false;
 
+  OhmmsAttributeSet H5attrib;
+  H5attrib.add (H5File, "href");
+  H5attrib.put(curRoot);
+  if(H5File!="")
+  {
+      ReadFromH5=true;
+      H5ReadOnce=true;
+  }
   if (myBasisSetFactory == 0)
   {//always create one, using singleton and just to access the member functions
     myBasisSetFactory = new BasisSetFactory(targetPtcl, targetPsi, ptclPool);
     myBasisSetFactory->setReportLevel(ReportLevel);
   }
-
   //check the basis set
   cur = curRoot->children;
   while (cur != NULL)//check the basis set
   {
     getNodeName(cname,cur);
-    if (cname == basisset_tag)
+    if (cname == basisset_tag || H5ReadOnce==true )
```

Comment:
@anbenali could you send me by email the case that the code will crash without this part of change.

Reply:
Done!